test/unittests/iotools/converters/oasis_to_bids/test_oasis_to_bids_utils.py

```diff
@@ -0,0 +1,172 @@
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+from clinica.iotools.converters.oasis_to_bids.oasis_to_bids_utils import (
+    create_sessions_dict,
+    write_sessions_tsv,
+)
+
+
+@pytest.fixture
+def build_clinical_data(tmp_path: Path) -> None:
+    (tmp_path / "clinical").mkdir()
+
+    df = pd.DataFrame(
+        {
+            "ID": ["OAS1_0001_MR1", "OAS1_0002_MR1"],
+            "M/F": ["F", "M"],
+            "Hand": ["R", "L"],
+            "Age": [74, 67],
+            "Educ": [2, 2],
+            "SES": [3, 3],
+            "MMSE": [29, 29],
+            "CDR": [0, 0],
+            "eTIV": [1344, 1344],
+            "nWBV": [0.704, 0.645],
+            "ASF": [1.306, 1.100],
+            "Delay": [float("nan"), float("nan")],
+        }
+    )
+    df.to_csv(tmp_path / "clinical" / "oasis_cross-sectional.csv", index=False)
+
+    # todo : future with excel
+
+
+@pytest.fixture
+def build_spec_sessions_success(tmp_path: Path) -> None:
+    (tmp_path / "spec").mkdir()
+    spec = pd.DataFrame(
+        {
+            "BIDS CLINICA": ["cdr_global", "MMS", "diagnosis", "foo"],
+            "ADNI": [np.nan, np.nan, np.nan, "foo"],
+            "OASIS": ["CDR", "MMSE", "CDR", np.nan],
+            "OASIS location": [
+                "oasis_cross-sectional.csv",
+                "oasis_cross-sectional.csv",
+                "oasis_cross-sectional.csv",
+                np.nan,
+            ],
+        }
+    )
+    spec.to_csv(tmp_path / "spec" / "sessions.tsv", index=False, sep="\t")
+
+
+@pytest.fixture
+def build_spec_sessions_error(tmp_path: Path) -> None:
+    (tmp_path / "spec").mkdir()
+    spec = pd.DataFrame(
+        {
+            "BIDS CLINICA": ["foo"],
+            "OASIS": ["foo"],
+            "OASIS location": [
+                "foo.csv",
+            ],
+        }
+    )
+    spec.to_csv(tmp_path / "spec" / "sessions.tsv", index=False, sep="\t")
+
+
+@pytest.fixture
+def build_bids_dir(tmp_path: Path) -> None:
+    (tmp_path / "BIDS" / "sub-OASIS10001" / "ses-M000").mkdir(parents=True)
+    (tmp_path / "BIDS" / "sub-OASIS10001" / "ses-M006").mkdir(parents=True)
+    (tmp_path / "BIDS" / "sub-OASIS10002" / "ses-M000").mkdir(parents=True)
```

Comment:
Your code works but is a bit surprising in the way you use fixtures.
Nothing wrong, but I would expect something more or less like this:

```python
@pytest.fixture
def bids_dir(tmp_path: Path) -> Path:
    bids_dir = tmp_path / "BIDS"
    _build_bids_dir(bids_dir)
    return bids_dir

def _build_bids_dir(bids_dir: Path) -> None:
    (bids_dir / "sub...").mkdir(....)
    ...

def test_create_bla_bla_bla(tmp_path, bids_dir: Path, ...):
    result = create_sessions_dict(clinical_data_dir, bids_dir, ...)
    assert result == ...
```

Does this make sense ? WDYT ?